doc: release/migration-notes/3.7: add bits about PCIe, I3C, UART and Xtensa (#75343)
zephyrproject-rtos/zephyr: doc/releases/release-notes-3.7.rst

* Xtensa

  * Added support to save/restore HiFi AudioEngine registers.

  * Added support to utilize MPU.

  * Added support to automatically generate interrupt handlers.

  * Added support to generate vector table at build time to be included in the linker script.

  * Added kconfig :kconfig:option:`CONFIG_XTENSA_BREAK_ON_UNRECOVERABLE_EXCEPTIONS` to guard

>     using break instruction for unrecoverable exceptions. The break instruction may result

Reviewer: ```suggestion
    using break instruction for unrecoverable exceptions. Without this, the break instruction may result
```
maybe better?

Author: Actually the kconfig enables using of break instruction. So I changed the wording a bit.

>     in infintie interrupt storm which may hinder debugging efforts.

Reviewer: ```suggestion
    in an infinite interrupt storm which may hinder debugging efforts.
```
article + typo

Author: Done.

>     this kconfig may result in an infintie interrupt storm which may hinder debugging efforts.

Reviewer 2: ```suggestion
    this kconfig may result in an infinite interrupt storm which may hinder debugging efforts.
```

Author: Done.

  * Fixed an issue where passing the 7th argument via syscall was handled incorrectly.

>   * Fixed an issue where :c:func:`arch_user_string_nlen` accessing unmapped memory results in

Reviewer 2: Nit, for tone consistency 

```suggestion
  * Fixed an issue where :c:func:`arch_user_string_nlen` accessing unmapped memory resulted in
```

Author: Done.

>     unrecoverable exception.

Reviewer: ```suggestion
    an unrecoverable exception.
```

Author: Done.

>     an unrecoverable exception.

Reviewer: ```suggestion
    in an unrecoverable exception.
```
or just "caused an urecoverable exception"

Author: Errr... stupid brain not working correctly today...

fixed.

* I3C

  * Added shell support for querying bus and CCC commands.

  * Added driver to support the I3C controller on NPCX.

>   * Improvements and bug fixes on ``i3c_mcux``, including handling of bus busy

Reviewer: ```suggestion
  * Improvements and bug fixes on ``i3c_mcux``, including handling the bus being busy
```
maybe better?

Author: Done.

>   * Improvements and bug fixes on ``i3c_mcux``, including handling the bus being busy

Reviewer 2: Here and elsewhere in your PR, it might be worth referencing drivers using their actual compatible and use `:dtcompatible:`

```suggestion
  * Improvements and bug fixes on :dtcompatible:`nxp,mcux-i3c`, including handling the bus being busy
```

Author: I have added the dtcompatibles. I want to keep the driver file names so it is easier to cross reference.
